### ISSUE TITLE
SC2028 false positive (using the correct model snippet)

### ISSUE THREAD

**Issue by vkucera** [koalaman/shellcheck#2486]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2028
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "\\t"

```

#### Here's what shellcheck currently says:
```
Line 2	SC2028: echo may not expand escape sequences. Use printf.
```
#### Here's what I wanted or expected to see:

No errors

**Comment by sigzero** [comment]
The comment now says:

```
[Line 2:](javascript:setPosition(2, 6))
echo "\\t"
     ^-- [SC2028](info): echo may not expand escape sequences. Use printf.
```

Since it is noting it as "information" and not a "warning" I find that acceptable.

**Comment by vkucera** [comment]
OK, but what is the point of showing this message for a code that is perfectly fine as opposed to a code that seems to have the SC2028 error?
Now shellcheck reports the same thing for both `echo "\\t"` and `echo "\t"` where the former is fine and the latter is suspicious.

**Comment by vkucera** [comment]
Besides, ShellCheck exits with 1 which makes it look like there is something to worry about which, to my understanding, is not the case with `\\t`.

**Comment by brother** [comment]
You can specify severity level to avoid info messages.

```
-S SEVERITY         --severity=SEVERITY        Minimum severity of errors to consider (error, warning, info, style)
```
Example

```
brother ~$ cat /tmp/test.sh
#!/bin/bash
echo "\\t"
brother ~$ shellcheck -S warning /tmp/test.sh
brother ~$ echo $?
0
brother ~$ shellcheck -S info /tmp/test.sh

In /tmp/test.sh line 2:
echo "\\t"
     ^---^ SC2028 (info): echo may not expand escape sequences. Use printf.

For more information:
  https://www.shellcheck.net/wiki/SC2028 -- echo may not expand escape sequen...
brother ~$ echo $?
1
```

**Comment by vkucera** [comment]
> You can specify severity level to avoid info messages.
> 
> ```
> -S SEVERITY         --severity=SEVERITY        Minimum severity of errors to consider (error, warning, info, style)
> ```
> 
> Example
> 
> ```
> brother ~$ cat /tmp/test.sh
> #!/bin/bash
> echo "\\t"
> brother ~$ shellcheck -S warning /tmp/test.sh
> brother ~$ echo $?
> 0
> brother ~$ shellcheck -S info /tmp/test.sh
> 
> In /tmp/test.sh line 2:
> echo "\\t"
>      ^---^ SC2028 (info): echo may not expand escape sequences. Use printf.
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC2028 -- echo may not expand escape sequen...
> brother ~$ echo $?
> 1
> ```

That does not address the point, which is that ShellCheck should report `echo "\t"` but not `echo "\\t"`.
Suppressing the info level entirely is not helping in any way.

**Comment by brother** [comment]
You stated
> Besides, ShellCheck exits with 1 which makes it look like there is something to worry about 

not everyone agrees with you on that. and in that casse there are severities to help you along. Just wanted you to know about it. The issue is still open - as you probably noted.

**Comment by aep-sunlife** [comment]
Even POSIX echo appears to support industry standard, C-style escape sequences:

https://pubs.opengroup.org/onlinepubs/9799919799/utilities/echo.html

This looks like a false positive to me, unless you can cite a specific exception (XSI?) in the latest Issue 8 edition of POSIX.
